functions/Read-DbaBackupHeader.ps1

```diff
@@ -0,0 +1,153 @@
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+https://dbatools.io/Read-DbaBackupHeader
+
+.EXAMPLE
+Read-DbaBackupHeader -SqlServer sql2016 -Path S:\backups\mydb\mydb.bak
+
+Logs into sql2016 using Windows authentication and reads the local file on sql2016, S:\backups\mydb\mydb.bak.
+	
+If you are running this command on a workstation and connecting remotely, remember that sql2016 cannot access files on your own workstation.
+
+.EXAMPLE
+Read-DbaBackupHeader -SqlServer sql2016 -Path \\nas\sql\backups\mydb\mydb.bak, \\nas\sql\backups\otherdb\otherdb.bak
+
+Logs into sql2016 and reads two backup files - mydb.bak and otherdb.bak. The SQL Server service account must have rights to read this file.
+	
+.EXAMPLE
+Read-DbaBackupHeader -SqlServer . -Path C:\temp\myfile.bak -Simple
+	
+Logs into the local worksation and shows simplified output about C:\temp\myfile.bak. The SQL Server service account must have rights to read this file.
+
+.EXAMPLE
+$backupinfo = Read-DbaBackupHeader -SqlServer . -Path C:\temp\myfile.bak
+$backupinfo.FileList
+	
+Displays detailed information about each of the datafiles contained in the backupset.
+
+.EXAMPLE
+Read-DbaBackupHeader -SqlServer . -Path C:\temp\myfile.bak -FileList
+	
+Also returns detailed information about each of the datafiles contained in the backupset.
+	
+#>
+	[CmdletBinding()]
+	Param (
+		[parameter(Mandatory = $true)]
+		[Alias("ServerInstance", "SqlInstance")]
+		[object]$SqlServer,
+		[object]$SqlCredential,
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[string[]]$Path,
+		[switch]$Simple,
+		[switch]$FileList
+	)
+	
+	BEGIN
+	{
+		
+		$server = Connect-SqlServer -SqlServer $sqlserver -SqlCredential $SqlCredential
+	}
+	
+	PROCESS
+	{
+		foreach ($file in $path)
+		{
+			$restore = New-Object Microsoft.SqlServer.Management.Smo.Restore
+			$device = New-Object -TypeName Microsoft.SqlServer.Management.Smo.BackupDeviceItem $file, "FILE"
+			$restore.Devices.Add($device)
+			
+			try
+			{
+				$allfiles = $restore.ReadFileList($server)
+			}
+			catch
+			{
+				if (!(Test-SqlPath -SqlServer $server -Path $file))
+				{
+					Write-Warning "File does not exist or access denied"
+				}
+				else
+				{
+					
+					Write-Warning "File list could not be determined. This is likely due to the file not existing, the backup version being incompatible or unsupported, connectivity issues or tiemouts with the SQL Server, or the SQL Server service account does not have access to the file share."
```

Comment:
'file not existing' is supposed to be a reason in the if-block
can it also be the reason for the else-block?
both test-sqlpath -eq $true and $false are possible for a non-existing file?